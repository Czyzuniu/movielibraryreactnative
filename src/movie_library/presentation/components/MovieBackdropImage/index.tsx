import React from 'react';
import MovieImage from '../MovieImage';

type Props = {
  imagePath?: string;
};

export default function MovieBackdropImage({imagePath}: Props) {
  return (
    <MovieImage
      resizeMode={'cover'}
      width={'full'}
      height={'3xs'}
      path={imagePath}
      alt={'movie-backdrop-image'}
    />
  );
}
