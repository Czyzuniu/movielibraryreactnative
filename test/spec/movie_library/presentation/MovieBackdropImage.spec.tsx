import MovieBackdropImage from '../../../../src/movie_library/presentation/components/MovieBackdropImage';
import {render} from '@testing-library/react-native';
import React from 'react';
import ContextWrapper from '../../../utils/ContextWrapper';

describe('MovieBackdropImage Tests', () => {
  it('should render and match snapshot', () => {
    const {toJSON} = render(
      <ContextWrapper>
        <MovieBackdropImage imagePath={'imagepath.jpg'} />
      </ContextWrapper>,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('should have the correct source uri', () => {
    const imagePath = 'imagepath.jpg';

    const {getByTestId} = render(
      <ContextWrapper>
        <MovieBackdropImage imagePath={imagePath} />
      </ContextWrapper>,
    );

    const image = getByTestId('TEST_IMAGE_' + imagePath);

    expect(image.props.source.uri).toBe(
      'https://mock-image-api.co.uk/w500/imagepath.jpg',
    );
  });
});
