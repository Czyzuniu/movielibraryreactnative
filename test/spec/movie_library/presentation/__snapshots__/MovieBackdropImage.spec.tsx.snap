// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieBackdropImage Tests should render and match snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <Image
    accessibilityLabel="movie-backdrop-image"
    alt="movie-backdrop-image"
    dataSet={Object {}}
    onError={[Function]}
    path="imagepath.jpg"
    resizeMode="cover"
    source={
      Object {
        "uri": "https://mock-image-api.co.uk/w500/imagepath.jpg",
      }
    }
    style={
      Object {
        "height": 224,
        "maxWidth": "100%",
        "width": "100%",
      }
    }
    testID="TEST_IMAGE_imagepath.jpg"
  />
</RNCSafeAreaProvider>
`;
